src/app/app.component.ts

```diff
@@ -12,19 +15,34 @@ export class MyApp {
 
   rootPage: any;
 
-  constructor(platform: Platform, statusBar: StatusBar, splashScreen: SplashScreen) {
+  constructor(
+    platform: Platform,
+    statusBar: StatusBar,
+    splashScreen: SplashScreen,
+    etherProvider : EtherProvider,
+    tokenProvider: TokenProvider,
+    notificationProvider: NotificationProvider
+  ) {
     platform.ready().then(() => {
       // Okay, so the platform is ready and our plugins are available.
       // Here you can do any higher level native things you might need.
       statusBar.styleDefault();
       splashScreen.hide();
 
+      //check if app have permission to show notifications
+      notificationProvider.checkPermission();
+
       //if an imported, scanned or created wallet exist
-      if(localStorage.getItem("isWallet") == "true") {
+      if(localStorage.getItem('isWallet') == 'true') {
+        let provider = etherProvider.get();
```

Comment:
below here we get the ether provider, the current wallet and the list of default tokens as an Array of objects and pass it to `setTokenListener()` to setup a listener for each token contract.